perl: improve matching pattern example, arg consistency, etc. (#8043)
tldr-pages/tldr: pages/common/perl.md

`perl -d {{script.pl}}`

> - Loo[p] over all lines of a file, editing them [i]n-place with a find/replace [e]xpression and saving the original file with a new extension:

Reviewer: ```suggestion
- Edit all file lines [i]n-place with a specific replacement [e]xpression and save a file with a new extension:
```

> `perl -p -i'.{{extension}}' -e 's/{{find}}/{{replace}}/g' {{filename}}`

Reviewer: ```suggestion
`perl -p -i'.{{extension}}' -e 's/{{regular_expression}}/{{replacement}}/g' {{path/to/file}}`
```

> - Run a multi-line replacement expression on a file, and save the result in a specific file:

Reviewer: ```suggestion
- Run a multi-line replacement [e]xpression on a file, and save the result in a specific file:
```

> - Run a multiline find/replace expression on a file, and save the result in another file:

Reviewer: ```suggestion
- Run a multi-line replacement expression on a file, and save the result in a specific file:
```

> `perl -p -e 's/{{foo\nbar}}/{{foobar}}/g' {{input_file}} > {{output_file}}`

Reviewer: ```suggestion
`perl -p -e 's/{{foo\nbar}}/{{foobar}}/g' {{path/to/input_file}} > {{path/to/output_file}}`
```

> - Run a regular expression on stdin, printing out matching [l]ines:

Reviewer: ```suggestion
- Run a regular [e]xpression on stdin, printing out matching [l]ines:
```

> `cat {{filename}} | perl -n -l -e 'print if /{{pattern}}/'`

Reviewer: ```suggestion
`cat {{path/to/file}} | perl -n -l -e 'print if /{{regular_expression}}/'`
```

We prefer `path/to/file` placeholder.

> - Run a regular expression on stdin, printing out only the first capture group for each matching [l]ine:

Reviewer: ```suggestion
- Run a regular [e]xpression on stdin, printing out only the first capture group for each matching [l]ine:
```

> `cat {{filename}} | perl -n -l -e 'print $1 if /{{before}}({{pattern}}){{after}}/'`

Reviewer: ```suggestion
`cat {{path/to/file}} | perl -n -l -e 'print $1 if /{{before}}({{regular_expression}}){{after}}/'`
```
